experimental_official_dockerfiles/master/Dockerfile

```diff
@@ -0,0 +1,73 @@
+################################################################################
+FROM ubuntu:18.04 as resourcebuilder
+################################################################################
+
+COPY install_packages.sh install_packages.sh
+COPY packages.build.txt packages.build.txt
+RUN /install_packages.sh /packages.build.txt
+
+# Build Python from source
+# TODO replace with github pinned url
+RUN curl https://pyenv.run | bash
+ENV PATH /root/.pyenv/shims:/root/.pyenv/bin:$PATH
+RUN pyenv install 3.6.9
+RUN pyenv install 3.7.7
+RUN pyenv install 3.8.2
+RUN pyenv install 3.9.1
+
+RUN git clone --depth=1 --single-branch --branch master https://github.com/tensorflow/tensorflow /tensorflow \
+        && cd /tensorflow/tensorflow/tools/ci_build/devtoolset \
+        && mv -t / fixlinks.sh build_devtoolset.sh rpm-patch.sh
+
+# Set up a sysroot for glibc 2.12 / libstdc++ 4.4 / devtoolset-7 in /dt7.
+RUN /build_devtoolset.sh devtoolset-7 /dt7 
+
+################################################################################
+FROM nvidia/cuda:11.0-base-ubuntu18.04 as devel
+################################################################################
+
+COPY --from=resourcebuilder /dt7 /dt7
+COPY --from=resourcebuilder /root/.pyenv /root/.pyenv
```

Comment:
Are we having multiple pyenvs in devel?

Reply:
I like the idea of keeping all the environments in a single image to simplify maintenance, especially if the TF team decides to use these for official builds. They each add ~300MB to the base image and ~some GB if installed with dependencies, though.

Reply:
Isn't better to trick like with `onlydefaultpy`? Other then the impact on download image size/Codespaces bootstrap having all the builds for all pyenvs isn't going to impact the remote cache?